Add docs for telemetry (metrics, traces and logging) (#845)
apache/polaris: site/content/in-dev/unreleased/telemetry.md

(port 8282 by default) under the path `/q/metrics`. For example, if the server is running on
localhost, the metrics can be accessed via http://localhost:8282/q/metrics.

[Micrometer]: https://quarkus.io/guides/telemetry-micrometer

Metrics can be scraped by Prometheus or any compatible metrics scraping server. See:
[Prometheus](https://prometheus.io) for more information.

Additional tags can be added to the metrics by setting the `polaris.metrics.tags.*` property. Each
tag is a key-value pair, where the key is the tag name and the value is the tag value. For example,
to add a tag `environment=prod` to all metrics, set `polaris.metrics.tags.environment=prod`. Many
tags can be added, such as below:

```properties
polaris.metrics.tags.service=polaris
polaris.metrics.tags.environment=prod
polaris.metrics.tags.region=us-west-2
```

Note that by default Polaris adds one tag: `application=Polaris`. You can override this tag by
setting the `polaris.metrics.tags.application=<new-value>` property.

## Traces

Traces are published using [OpenTelemetry]. 

[OpenTelemetry]: https://quarkus.io/guides/opentelemetry-tracing

> In order for the server to enable OpenTelemetry and publish its traces, the
> `quarkus.otel.exporter.otlp.traces.endpoint` property _must_ be defined. Its value must be a valid
> collector endpoint URL, with either `http://` or `https://` scheme. The collector must talk the

Reviewer: ```suggestion
To enable OpenTelemetry and publish traces for Polaris, set a valid collector endpoint URL with `http://` or `https://` as the server property `quarkus.otel.exporter.otlp.traces.endpoint`. The collector must talk the
```

OpenTelemetry protocol (OTLP) and the port must be its gRPC port (by default 4317), e.g.
"http://otlp-collector:4317". _If this property is not set, the server will not publish traces._

> Alternatively, it's possible to forcibly disable OpenTelemetry at runtime by setting the following
> property: `quarkus.otel.sdk.disabled=true`.

Reviewer: ```suggestion
You can disable OpenTelemetry at runtime by setting `quarkus.otel.sdk.disabled=true`.
```
